sql/core/src/test/scala/org/apache/spark/sql/internal/CatalogSuite.scala

```diff
@@ -700,7 +700,7 @@ class CatalogSuite extends SharedSparkSession with AnalysisTest with BeforeAndAf
     val description = "this is a test table"
 
     withTable("t") {
-      withTempDir { dir =>
+      withTempDir(prefix = "test%prefix") { dir =>
```

Comment:
I'm not sure we need this new util function with `prefix` parameter, we can just do
```
withTempDir { baseDir =>
  val dir = new File(baseDir, "test%prefix")
```

Reply:
That would also work, but I think is useful to have it since we should expand more on testing of paths with special chars. In the future we could also change the default value, so this helper will be needed to avoid the special chars.

Reply:
It's orthogonal from this bug fix, isn't it? I'd recommend to proceed that test utility contribution as an independent PR, @cstavr . We can review separately.

> That would also work, but I think is useful to have it since we should expand more on testing of paths with special chars. In the future we could also change the default value, so this helper will be needed to avoid the special chars.

Reply:
ACK.  I removed the test util.